wgpu-core/src/device/resource.rs

```diff
@@ -272,6 +282,25 @@ impl Device {
         let alignments = adapter.raw.capabilities.alignments.clone();
         let downlevel = adapter.raw.capabilities.downlevel.clone();
 
+        #[cfg(feature = "indirect-validation")]
+        let indirect_validation = if downlevel
+            .flags
+            .contains(wgt::DownlevelFlags::INDIRECT_EXECUTION)
+        {
+            match crate::indirect_validation::IndirectValidation::new(
+                raw_device.as_ref(),
+                &desc.required_limits,
+            ) {
+                Ok(indirect_validation) => Some(indirect_validation),
+                Err(e) => {
+                    log::error!("indirect-validation error: {e:?}");
+                    return Err(CreateDeviceError::Internal);
```

Comment:
issue: `CreateDeviceError` isn't even _used_ any more, and seems like it should have been removed as of <https://github.com/gfx-rs/wgpu/pull/6229> (CC @teoxoy as the author), so this addition of `CreateDeviceError::Internal` no longer makes sense.

issue: We have lots of other compilation errors after rebasing because we no longer `impl From<hal::DeviceError> for wgpu_core::DeviceError` (which is a good thing, also from <https://github.com/gfx-rs/wgpu/pull/6229>). We need to use `wgpu_core::DeviceError::from_hal`. This is appropriate, rather than `Device::handle_hal_error`, because the context is that we're trying to create a `Device` in the first place.

suggestion: I have resolved both of the above as follows with the force-pushed b6b2faed:

```diff
--- a/wgpu-core/src/device/resource.rs
+++ b/wgpu-core/src/device/resource.rs
@@ -283,7 +283,7 @@
                 Ok(indirect_validation) => Some(indirect_validation),
                 Err(e) => {
                     log::error!("indirect-validation error: {e:?}");
-                    return Err(CreateDeviceError::Internal);
+                    return Err(DeviceError::Lost);
                 }
             }
         } else {
--- a/wgpu-core/src/indirect_validation.rs
+++ b/wgpu-core/src/indirect_validation.rs
@@ -107,7 +107,7 @@
             unsafe { device.create_shader_module(&hal_desc, hal_shader) }.map_err(|error| {
                 match error {
                     hal::ShaderError::Device(error) => {
-                        CreateShaderModuleError::Device(error.into())
+                        CreateShaderModuleError::Device(DeviceError::from_hal(error))
                     }
                     hal::ShaderError::Compilation(ref msg) => {
                         log::error!("Shader error: {}", msg);
@@ -133,7 +133,7 @@
         let dst_bind_group_layout = unsafe {
             device
                 .create_bind_group_layout(&dst_bind_group_layout_desc)
-                .map_err(DeviceError::from)?
+                .map_err(DeviceError::from_hal)?
         };
 
         let src_bind_group_layout_desc = hal::BindGroupLayoutDescriptor {
@@ -153,7 +153,7 @@
         let src_bind_group_layout = unsafe {
             device
                 .create_bind_group_layout(&src_bind_group_layout_desc)
-                .map_err(DeviceError::from)?
+                .map_err(DeviceError::from_hal)?
         };
 
         let pipeline_layout_desc = hal::PipelineLayoutDescriptor {
@@ -171,7 +171,7 @@
         let pipeline_layout = unsafe {
             device
                 .create_pipeline_layout(&pipeline_layout_desc)
-                .map_err(DeviceError::from)?
+                .map_err(DeviceError::from_hal)?
         };
 
         let pipeline_desc = hal::ComputePipelineDescriptor {
@@ -188,7 +188,7 @@
         let pipeline =
             unsafe { device.create_compute_pipeline(&pipeline_desc) }.map_err(|err| match err {
                 hal::PipelineError::Device(error) => {
-                    CreateComputePipelineError::Device(error.into())
+                    CreateComputePipelineError::Device(DeviceError::from_hal(error))
                 }
                 hal::PipelineError::Linkage(_stages, msg) => {
                     CreateComputePipelineError::Internal(msg)
@@ -208,7 +208,7 @@
             memory_flags: hal::MemoryFlags::empty(),
         };
         let dst_buffer =
-            unsafe { device.create_buffer(&dst_buffer_desc) }.map_err(DeviceError::from)?;
+            unsafe { device.create_buffer(&dst_buffer_desc) }.map_err(DeviceError::from_hal)?;
 
         let dst_bind_group_desc = hal::BindGroupDescriptor {
             label: None,
@@ -230,7 +230,7 @@
         let dst_bind_group = unsafe {
             device
                 .create_bind_group(&dst_bind_group_desc)
-                .map_err(DeviceError::from)
+                .map_err(DeviceError::from_hal)
         }?;
 
         Ok(Self {
@@ -272,7 +272,7 @@
         unsafe {
             device
                 .create_bind_group(&hal_desc)
-                .map_err(DeviceError::from)
+                .map_err(DeviceError::from_hal)
         }
     }
 
```

Reply:
Thanks!